firefox_ui_tests/remote/security/test_safe_browsing_warning_pages.py

```diff
@@ -0,0 +1,116 @@
+        get_me_out_of_here_button = self.marionette.find_element(By.ID, "getMeOutButton")
+
+        # This isn't clickable by the time we get here and needs a delay.
+        time.sleep(1)
+        get_me_out_of_here_button.click()
+
+        homepage = self.prefs.get_pref('browser.startup.homepage',
+                                       interface='nsIPrefLocalizedString')
+        self.wait_for_condition(lambda mn: homepage in mn.get_url())
+
+    def check_report_button(self, unsafe_page):
+        report_button = self.marionette.find_element(By.ID, "reportButton")
+
+        # This isn't clickable by the time we get here and needs a delay.
+        time.sleep(1)
+        report_button.click()
+
+        with self.marionette.using_context('chrome'):  # required by execute_script
+            if 'its-a-trap' in unsafe_page:
+                # mozmill: utils.formatUrlPref("app.support.baseURL") + "phishing-malware"
+                # the code commented out below produces this:
+                # https://support.mozilla.org/1/firefox/38.0a1/Darwin/en-US/phishing-malware
+                # which fails to match the page we're sent to...
+                # url = self.marionette.execute_script("""
+                #    Cu.import("resource://gre/modules/Services.jsm");
+                #    locale = Services.prefs.getCharPref("general.useragent.locale", "");
+                #    return (Services.urlFormatter.formatURLPref("app.support.baseURL")
+                #                                               + "phishing-malware");
+                #    """)
+                url = 'https://support.mozilla.org'
+
+            else:
+                url = self.marionette.execute_script("""
+                   Cu.import("resource://gre/modules/Services.jsm");
+                   locale = Services.prefs.getCharPref("general.useragent.locale", "");
+                   return (Services.urlFormatter.formatURLPref(
+                            "browser.safebrowsing.malware.reportURL") + arguments[0]);
+                   """, script_args=[unsafe_page])
+
+        self.wait_for_condition(lambda mn: url in mn.get_url())
+
+    def check_ignore_warning_button(self, unsafe_page):
+        ignore_warning_button = self.marionette.find_element(By.ID, "ignoreWarningButton")
+
+        # This isn't clickable by the time we get here and needs a delay.
+        time.sleep(1)
+        ignore_warning_button.click()
+
+        def find_main_feature_el(mn):
+            try:
+                mn.find_element("id", "main-feature")
+                return True
+            except:
+                return False
+
+        self.wait_for_condition(find_main_feature_el)
+        self.assertRaises(NoSuchElementException, self.marionette.find_element,
+                          'id', 'ignoreWarningButton')
+        self.assertEquals(self.marionette.get_url(), unsafe_page)
+        self.perms.remove('www.itisatrap.org', 'safe-browsing')
```

Comment:
No, it needs to run between the Phishing URL test and the Malware URL test.

It fits better in the main test function. I'll add it there and remote the tearDown.